roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/util/ConfigParserTest.java

```diff
@@ -117,13 +121,95 @@ public void configParserTest_generateModel() throws ConfigParserException {
+    }
+
+    @Test
+    public void configParserTest_structureCorrect() throws ConfigParserException {
+        Experiment exp = new Experiment("Experiment");
+        IRoundaboutStructure roundaboutStructure = configParser.generateRoundaboutStructure(roundAboutConfig, exp);
+
+        for (IStreetConnector connector : roundaboutStructure.getStreetConnectors()) {
+            Integer exitSectionId = null;
+            Integer nextTrackSectionId = null;
+            for (IConsumer section : connector.getNextSections()) {
+                if (section.toString().contains("exit")) {
+                    exitSectionId = extractSectionId(section.toString());
+                } else if (section.toString().contains("track")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (nextTrackSectionId != null && nextTrackSectionId != id) {
+                        assertTrue("tracks from different sections on next connector", false);
+                    }
+                    nextTrackSectionId = id;
+                } else {
+                    assertTrue("only tracks and exit on next connector", false);
+                }
+            }
+
+            Integer entrySectionId = null;
+            Integer previousTrackSectionId = null;
+            for (IProducer section : connector.getPreviousSections()) {
+                if (section.toString().contains("entry")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (entrySectionId != null && entrySectionId != id) {
+                        assertTrue("entries from different sections on next connector", false);
+                    }
+                    entrySectionId = id;
+                } else if (section.toString().contains("track")) {
+                    Integer id = extractSectionId(section.toString());
+                    if (previousTrackSectionId != null && previousTrackSectionId != id) {
+                        assertTrue("tracks from different sections on next connector", false);
+                    }
+                    previousTrackSectionId = id;
+                } else {
+                    assertTrue("only tracks and entries on previous connector", false);
+                }
+            }
+
+            assertTrue("only tracks from neighbour sections", areNeighbourSections(previousTrackSectionId, nextTrackSectionId, roundaboutStructure.getStreetConnectors().size()));
+            if (exitSectionId != null) { // TODO fix after connector implementation
+                assertEquals("exit and previous tracks from same section", exitSectionId, previousTrackSectionId);
+            }
+            if (entrySectionId != null) { // TODO fix after connector implementation
+                assertEquals("entries and next tracks from same section", entrySectionId, nextTrackSectionId);
+            }
+        }
+    }
+
+    private Integer extractSectionId(String value) {
```

Comment:
maybe the section itself should be able to return a its id?

Reply:
it's only needed for testing, not in the simulation